src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -1483,4 +1503,77 @@ std::string SQLiteDBEngine::getSelectAllQuery(const std::string& table,
     }
 
     return retVal;
-} 
\ No newline at end of file
+} 
+
+std::string SQLiteDBEngine::buildDeleteRelationTrigger(const nlohmann::json& data,
+                                                       const std::string&    baseTable)
+{
+    const constexpr auto DELETE_POSTFIX{"_delete"};
+
+    auto sqlDelete
+    {
+        "CREATE TRIGGER IF NOT EXISTS " + baseTable + DELETE_POSTFIX + " BEFORE DELETE ON " + baseTable
+    };
+
+    sqlDelete.append(" BEGIN ");
+    for (const auto& jsonValue : data.at("relationed_tables"))
+    {
+        sqlDelete.append("DELETE FROM "+ jsonValue.at("table").get<std::string>() + " WHERE ");
+        for (const auto& match : jsonValue.at("field_match").items())
+        {
+            sqlDelete.append(match.key());
+            sqlDelete.append(" = OLD.");
+            sqlDelete.append(match.value());
+            sqlDelete.append(" AND ");
+        }
+        sqlDelete = sqlDelete.substr(0, sqlDelete.size()-5);
+        sqlDelete.append(";");
+    }
+    sqlDelete.append("END;");
+    return sqlDelete;
+}
+
+std::string SQLiteDBEngine::buildUpdateRelationTrigger(const nlohmann::json&            data,
+                                                       const std::string&               baseTable,
+                                                       const std::vector<std::string>   primaryKeys)
+{
+    const constexpr auto UPDATE_POSTFIX{"_update"};
+
+    std::string sqlUpdate;
+
+    sqlUpdate.append("CREATE TRIGGER IF NOT EXISTS " + baseTable + UPDATE_POSTFIX + " BEFORE UPDATE OF ");
```

Comment:
Maybe you could initialize the sqlUpdate with the "append" data at once. I mean:

std::string sqlUpdate
{
   "CREATE TRIGGER IF NOT EXISTS " + baseTable + UPDATE_POSTFIX + " BEFORE UPDATE OF "
};